### PR TITLE
Optimization, error handlers

### PR DESCRIPTION
**Pull request by Alexcitten**
**Possible error handlers were added, optimization was done, and a bit of beauty was added.**

The initialization of Arweave was moved outside of the `deployContract` function, since there is no need to initialize it several times.
Then created a separate `deployContract` function to handle the deployment process. Renamed some variables for clarity. And also added error handling for the deploy method

I used ES6 to import individual functions, and simplified operators to check input and error handling.
Created a separate functionMap object to map input functions to the corresponding function implementations.
Useful to have error handling when the input function is not in the function map.
Renamed the change-price function to `changePrice` to follow the standard naming conventions in the camel case, I understand that such "cutting" of code is not necessary, but why not.

Used destructive assignment to get the input and caller properties from the action object, and the default value syntax to set the target variable to an undefined value if it is not present in the input object looks nice.

I also changed the `ContractAssert` condition so that it is not executed when the target is false, moved the calculation of the balance variable after the `ContractAssert` condition to avoid unnecessary calculations when the target is not present.
Simplified the balance calculation with the `||` operator so that the default value is 0 when the target is not present in the balance object.

I used destructive assignment to get the input and caller properties from the action object, and changed the second ContractAssert condition so that it doesn't execute if target is false.
A good solution is to move `balances[caller] -= qty` before `balances[target] += qty` to avoid double-counting when caller and target are the same.
Added check to remove caller from balances when caller no longer owns tokens.

### CODE REVIEW

**Review by angrymouse — CHANGES_REQUESTED** [review]
Can you also add contract code address to output? It's useful property

File: src/contract.js [hunk 1]
```diff
@@ -1,19 +1,32 @@
+import transfer from "./functions/transfer.js";
```
Comment (angrymouse) [inline src/contract.js]: Will imports work with esbuild's IIFE setup?
It can potentially break execution

File: src/functions/balance.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
 module.exports = async function (state, action) {
```
Comment (angrymouse) [inline src/functions/balance.js]: if you use ESM import, use ESM export as well, or it doesn't work

**Comment by Alexcitten** [comment]
Yes, you are correct. I should have thought about that, and checked it out.
Kind of fixed it.

File: deploy.js [hunk 3]
```diff
@@ -31,7 +31,9 @@ async function deployContract() {
   const { contractTxId, srcTxId } = deployResult;
   console.log(`Deployment completed.\nDeployer: ${walletAddress}\nContract address: ${contractTxId}\nCode address: ${srcTxId}`);
 
-  return deployResult;
+  // Displaying the deployed contract code
+  const contractCode = await arweave.transactions.getData(srcTxId, { decode: true, string: true });
```
Comment (angrymouse) [inline deploy.js]: You don't have to re-download code... What I meant is showing Transaction ID of code that you've uploaded to arweave

**Comment by Alexcitten** [comment]
I hope I've done everything right now.
A description of each improvement can be found in the comet description

**Pull request merged**